whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisRestServiceDoSTestIntegration.java

```diff
@@ -0,0 +1,249 @@
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+
+@Tag("IntegrationTest")
+public class WhoisRestServiceDoSTestIntegration extends AbstractIntegrationTest {
+
+    @Autowired
+    private WhoisObjectMapper whoisObjectMapper;
+
+    private static final RpslObject OWNER_MNT = RpslObject.parse("" +
+            "mntner:      OWNER-MNT\n" +
+            "descr:       Owner Maintainer\n" +
+            "admin-c:     TP1-TEST\n" +
+            "upd-to:      noreply@ripe.net\n" +
+            "auth:        MD5-PW $1$d9fKeTr2$Si7YudNf4rUGmR71n/cqk/ #test\n" +
+            "auth:        SSO person@net.net\n" +
+            "mnt-by:      OWNER-MNT\n" +
+            "source:      TEST");
+
+    private static final Integer PENULTIMATE_OK_UPDATE_POSITION = 8;
+
+    private static final Integer PENULTIMATE_OK_LOOKUP_POSITION = 48;
+
+    private static final Integer MAXIMUM_UPDATE_REQUESTS_ALLOWED_PER_SECOND = 10;
+
+    private static final Integer MAXIMUM_LOOKUP_REQUESTS_ALLOWED_PER_SECOND = 50;
+
+    private static final Integer SECONDS_NEEDED_TO_FREE_IP = 1;
+
+    @BeforeAll
+    public static void beforeClass() {
+        System.setProperty("dos.filter.enabled", "true");
+    }
+
+    @BeforeEach
+    public void setup() {
+        databaseHelper.addObject("person: Test Person\nnic-hdl: TP1-TEST");
+        databaseHelper.addObject(OWNER_MNT);
+        testDateTimeProvider.setTime(LocalDateTime.parse("2001-02-04T17:00:00"));
+    }
+
+    // lookup
+    @Test
+    public void multiple_lookup_per_second_then_429_too_many_requests() throws InterruptedException {
+        final Response response = RestTest.target(getPort(), "whois/test/mntner/OWNER-MNT?clientIp=10.20.30.40")
+                .request()
+                .get();
+        assertThat(HttpStatus.OK_200, is(response.getStatus()));
+
+        final Invocation.Builder lookupRequest = RestTest.target(getPort(), "whois/test/mntner/OWNER-MNT?clientIp=10.20.30.40").request();
+
+        // Simulate a DoS attack by sending many GET requests in a short time
+        final List<Integer> responsesCodes = IntStream.range(0, MAXIMUM_LOOKUP_REQUESTS_ALLOWED_PER_SECOND)
+                .mapToObj(lookupCount -> lookupRequest.get())
+                .map(Response::getStatus)
+                .toList();
+
+        assertThat(HttpStatus.OK_200, is(responsesCodes.get(PENULTIMATE_OK_LOOKUP_POSITION)));
+        assertThat(HttpStatus.TOO_MANY_REQUESTS_429, is(responsesCodes.getLast()));
+
+        TimeUnit.SECONDS.sleep(SECONDS_NEEDED_TO_FREE_IP); // Free the IP after one second
```

Comment:
instead of sleep , can we set the dateTime ? This will make test faster and consistent

Reply:
I think that the purpose of testDateTImeProvider is different.
It can not be used here for making a delay. This seems to be for managing RPSL object instead timestamp.

I tried with 
`testDateTimeProvider.setTime(LOCAL_DATE_TIME.plusSeconds(1));`

Instead the sleep but it doesn't work. Then I checked JdbcSerialDoaIntegration and I see that the usage is different

Reply:
Thanks for checking